src/Controller/ApiController.php

```diff
@@ -20,33 +15,20 @@
  */
 class ApiController extends AbstractController
 {
-    const RADIO_CLASSES = [
-        MetalOnly::class,
-        RadioGalaxy::class,
-        RauteMusik::class,
-        StarFm::class,
-        TechnoBase::class,
-    ];
-
     /** @var array */
-    private static $radios;
+    private $radios = [];
 
-    public function __construct()
+    public function addRadio(string $radio)
     {
-        if (!static::$radios) {
-            /** @var AbstractRadioStream $radioClass */
-            foreach (self::RADIO_CLASSES as $radioClass) {
-                static::$radios[$radioClass::getRadioName()] = $radioClass;
-            }
-        }
+        $this->radios[$radio::getRadioName()] = $radio;
```

Comment:
I'd recommend adding a kernelView listener to handle serialization

Reply:
The `json()` from the `AbstractController` (via `ControllerTrait`) that I'm currently using basically does everything I need. If I would use the `kernelView` listener I would have to create the json myself, or just do the same thing that `json()` is already doing for me, so I think currently it's ok to just keep on using `json()` but thanks anyway :+1: I will keep the `kernelView` listener in mind, maybe it will come in handy if I will have more complex requirements in the future :)